packages/front-end/components/Settings/CustomFieldModal.tsx

```diff
@@ -0,0 +1,190 @@
+import { useForm } from "react-hook-form";
+import { CustomField, CustomFieldTypes } from "back-end/types/organization";
+import uniqid from "uniqid";
+import React from "react";
+import { useAuth } from "@/services/auth";
+import { useCustomFields } from "@/services/experiments";
+import { useUser } from "@/services/UserContext";
+import { useDefinitions } from "@/services/DefinitionsContext";
+import MultiSelectField from "@/components/Forms/MultiSelectField";
+import Modal from "../Modal";
+import Field from "../Forms/Field";
+import Toggle from "../Forms/Toggle";
+import SelectField, { GroupedValue, SingleValue } from "../Forms/SelectField";
+import track from "../../services/track";
+
+export default function CustomFieldModal({
+  existing,
+  close,
+  onSuccess,
+}: {
+  existing: Partial<CustomField>;
+  close: () => void;
+  onSuccess?: () => void;
+}) {
+  const { project, projects } = useDefinitions();
+  const { apiCall } = useAuth();
+  const { userId } = useUser();
+  const form = useForm<Partial<CustomField>>({
+    defaultValues: {
+      id: existing.id || uniqid("field_"),
+      name: existing.name || "",
+      values: existing.values || "",
+      type: existing.type || "text",
+      section: "experiment", // not supporting features yet
+      projects: existing.projects || [project] || [],
+      required: existing.required ?? false,
+      dateCreated:
+        existing.dateCreated || new Date().toISOString().substr(0, 16),
+      active: true,
+      index: true,
+    },
+  });
+
+  const customFields = useCustomFields() ?? [];
+
+  const fieldOptions = [
+    "text",
+    "textarea",
+    "markdown",
+    "enum",
+    "multiselect",
+    "boolean",
+  ];
+
+  const availableProjects: (SingleValue | GroupedValue)[] = projects
+    .slice()
+    .sort((a, b) => (a.name > b.name ? 1 : -1))
```

Comment:
I don't think the other projects filters have sort (maybe they should?). But this one is not case insensitive.